src/main/resources/com/google/api/codegen/go/example.snip

```diff
@@ -22,24 +22,110 @@
     }
 
     @join method : view.apiMethods
-        @if method.isPageStreaming
+        @switch view.getTrue
+        @case method.isBidiStreaming
+            {@bidiStreamingMethod(view, method)}
+        @case method.isRequestStreaming
+            {@requestStreamingMethod(view, method)}
+        @case method.isResponseStreaming
+            {@responseStreamingMethod(view, method)}
+        @case method.isPageStreaming
             {@pageStreamingMethod(view, method)}
-        @else
-            @if method.hasReturnValue
-                {@simpleMethod(view, method)}
-            @else
-                {@emptyReturnMethod(view, method)}
-            @end
+        @case method.hasReturnValue
+            {@simpleMethod(view, method)}
+        @default
+            {@emptyReturnMethod(view, method)}
```

Comment:
I thought it looked odd but found it strangely familiar, then I realized: https://tour.golang.org/flowcontrol/11